exonum/src/node/consensus.rs

```diff
@@ -613,16 +680,31 @@ impl NodeHandler {
     /// Handles external boxed transaction. Additionally transaction will be broadcast to the
     /// Node's peers.
     #[cfg_attr(feature = "cargo-clippy", allow(clippy::needless_pass_by_value))]
-    pub fn handle_incoming_tx(&mut self, msg: Verified<AnyTx>) {
+    pub(crate) fn handle_incoming_tx(&mut self, msg: Verified<AnyTx>) {
         trace!("Handle incoming transaction");
+        // Check transaction before handling it to avoid the broadcasting.
+        // Reaction for an incorrect tx here and in `handle_tx` is different:
+        // here we simply ignore it without notifying the network, but if the
+        // transaction is received from the network, we have to deal with it
+        // and maybe even panic (if most of nodes will approve a block with such a
+        // transaction).
+        if let Err(error) = self.blockchain.check_tx(&msg) {
```

Comment:
Off-topic: I wonder if it makes sense to avoid this check in favor of checks in the places where the external transactions are accepted. For example, it could lead to more informative errors in the explorer's `POST transactions` endpoint. Note that per #1565, Rust services can no longer send transactions to any recipient service other than themselves, although it's not true (yet?) for the Java runtime, and there is a possibility to use `Blockchain::sender()` directly.